lib/matplotlib/lines.py

```diff
@@ -353,14 +353,8 @@ def __init__(self, xdata, ydata,
         self._linestyles = None
         self._drawstyle = None
         self._linewidth = linewidth
-
-        # scaled dash + offset
-        self._dashSeq = None
-        self._dashOffset = 0
-        # unscaled dash + offset
-        # this is needed scaling the dash pattern by linewidth
-        self._us_dashSeq = None
-        self._us_dashOffset = 0
+        self._us_dash_pattern = (0, None)  # offset, dash (unscaled)
```

Comment:
I‘m inclined to use a namedtuple instead of the comment for semantic clarity.

Reply:
At that point, should it be a dataclass so that it can eventually carry documentation/the linestyle docs can be pushed there?

Reply:
At that point, should it be a dataclass so that it can eventually carry documentation/the linestyle docs can be pushed there?

Reply:
You *could* write `self._dash_pattern = DashPattern(0, None)` but I would argue that this is actually worse and still needs a comment (or `DashPattern(offset=0, dashes=None)` to not need a comment), as it is *less* clear that we're just taking two values and not doing anything fancy with them (whereas a constructor could do arbitrary things with its arguments).
Additionally, we'd never actually use the `.offset` and `.dashes` accessors, which removes the main benefit of having named fields (well, _scale_dashes could take a packed dashpattern rather than an unpacked dashpattern as parameter, but that doesn't seem like a big deal).